src/ebpf_yaml.cpp

```diff
@@ -257,13 +257,17 @@ static vector<T> vector_of(const std::vector<uint8_t>& bytes) {
     return {(T*)data, (T*)(data + size)};
 }
 
-template <typename T>
+template <typename TS, typename TU>
 void add_stack_variable(std::set<std::string>& more, int& offset, const std::vector<uint8_t>& memory_bytes) {
-    T value;
-    memcpy(&value, memory_bytes.data() + offset + memory_bytes.size() - EBPF_STACK_SIZE, sizeof(T));
-    more.insert("s[" + std::to_string(offset) + "..." + std::to_string(offset + sizeof(T) - 1) +
-                "].value=" + std::to_string(value));
-    offset += sizeof(T);
+    TS svalue;
+    TU uvalue;
+    memcpy(&svalue, memory_bytes.data() + offset + memory_bytes.size() - EBPF_STACK_SIZE, sizeof(TS));
+    memcpy(&uvalue, memory_bytes.data() + offset + memory_bytes.size() - EBPF_STACK_SIZE, sizeof(TU));
+    more.insert("s[" + std::to_string(offset) + "..." + std::to_string(offset + sizeof(TS) - 1) +
+                "].svalue" + "=" + std::to_string(svalue));
+    more.insert("s[" + std::to_string(offset) + "..." + std::to_string(offset + sizeof(TU) - 1) +
+                "].uvalue" + "=" + std::to_string(uvalue));
+    offset += sizeof(TS);
```

Comment:
`TU` can be replaced with [`make_unsigned<TS>::type`](https://en.cppreference.com/w/cpp/types/make_unsigned), which by the way documents the intention and checks it.

Reply:
Done